### PR TITLE
chore: caching.feature integration tests

### LINKED ISSUE #243: Implement caching integration tests
The testing harness has introduced a [caching suite](https://github.com/open-feature/test-harness/blob/main/features/caching.feature), tests for this should be implemented.

### PR DESCRIPTION
**Pull request by skyerus**
## This PR

Implements caching test suite.

### Related Issues

Fixes #243

### CODE REVIEW

**Comment by skyerus** [comment]
Converted to draft until v0.5.5 of flagd-provider is released

**Comment by skyerus** [comment]
Running into some confusing behaviour here, the flakey results of tests are implying that the flagd provider is exhibiting asynchronous behaviour where it ought not to be.
This is one of the tests that is passing/failing intermittently:
> Given a boolean flag with key "boolean-flag" is evaluated with details and default value "false"
    When a boolean flag with key "boolean-flag" is evaluated with details and default value "false"
    Then the resolved boolean details reason should be "CACHED"

Stepping through the code, my understanding is that a boolean evaluation stores the value in cache synchronously in the first call then when the second call is invoked the value is retrieved from the cache. Either my understanding about the synchronous nature is wrong or there is some kind of disconnection happening with the event stream that is causing the cache to be ignored (by design).

Will debug further.

**Comment by skyerus** [comment]
This exposure of a blocking mechanism until the event stream is alive fixes the flakiness: https://github.com/open-feature/java-sdk-contrib/pull/204

**Review by toddbaert — DISMISSED** [review]

File: src/test/java/dev/openfeature/sdk/integration/StepDefinitions.java [hunk 1]
```diff
@@ -12,24 +15,35 @@
...
+import io.cucumber.java.en.And;
+
+import java.io.File;
+import java.io.FileWriter;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
 
 public class StepDefinitions {
```
Comment (Kavindu-Dodan) [inline src/test/java/dev/openfeature/sdk/integration/StepDefinitions.java:31]: Let's annotate this with `@Slf4j` get logger enabled

File: src/test/java/dev/openfeature/sdk/integration/StepDefinitions.java [hunk 2]
```diff
@@ -39,15 +53,34 @@ public class StepDefinitions {
...
-    public static void setup() {
-        // TODO: when the FlagdProvider is updated to support caching, we might need to disable it here for this test to work as expected.
-        FlagdProvider provider = new FlagdProvider();
+    @BeforeAll
+    public static void init() {
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            flagConfig = mapper.readTree(new File(testingFlagsPath));
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (Kavindu-Dodan) [inline src/test/java/dev/openfeature/sdk/integration/StepDefinitions.java:65]: let's use logger with `@Slf4j` annotation  for all such places

Comment (skyerus) [reply src/test/java/dev/openfeature/sdk/integration/StepDefinitions.java]: thanks, done

File: src/test/java/dev/openfeature/sdk/integration/StepDefinitions.java (hunk 1 above)
Comment (skyerus) [reply src/test/java/dev/openfeature/sdk/integration/StepDefinitions.java:33]: thanks, done

**Review by toddbaert — APPROVED** [review]

**Review by Kavindu-Dodan — DISMISSED** [review]

**Comment by toddbaert** [comment]
@skyerus The caching tests here appear to be hanging. I've re-run them a few times.

**Comment by toddbaert** [comment]
I'll look into what's going on here.

**Comment by justinabrahms** [comment]
@toddbaert Is this still something to look into? Or close?

**Comment by beeme1mr** [comment]
This PR appears to be stale. Please reopen if you would like to continue working on it.

**Pull request closed without merging**